src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -18,7 +19,7 @@ internal sealed partial class XmlSqlBinaryReader : XmlReader, IXmlNamespaceResol
         internal static readonly Type TypeOfObject = typeof(object);
         internal static readonly Type TypeOfString = typeof(string);
 
-        private static volatile Type[] s_tokenTypeMap = null;
+        private static volatile Type?[]? s_tokenTypeMap;
```

Comment:
possible, will check - I have removed `= null` for some reason but can't remember from the top of my head why :smile:

Reply:
I agree on outer being not null (except before you create first instance but it doesn't matter since no one is using it then) but inner one can be null since array is much larger than number of entries so changing to `Type?[]`